### ISSUE TITLE
Toolchain-xtensa-esp32s2 for linux_aarch64

### ISSUE THREAD

**Issue by maedball** [platformio/platform-espressif32#699]
Hiho,

i would be super helpful for all Homeassistant Blue users with ESP home to have the toolchain-xtensa-esp32s2 available in linux_aarch64.

Would it be possible to make this available?

Thanks

**Comment by homonto** [comment]
+1 from me - I have a bunch of ESP32-S2 and I cannot use them with ESPHome that is on Raspberry Pi

**Comment by hmax42** [comment]
the same problem affects esp32-c3

**Comment by DonBosserez** [comment]
+1 for me - also running HASS on Odroid hardware.

**Comment by GNAKACH** [comment]
Hi Same problem for me; Same configuration: Homme assistant running on a raspberry Pi 4 with Home assistant add on.
It is really a fantastic tool.
alas, I cannot use my ESP32-S2 as the platform is missing:(
I hope the implementation arrives soon ;)

**Comment by maedball** [comment]
> > Hi Same problem for me; Same configuration: Homme assistant running on a raspberry Pi 4 with Home assistant add on.
> > It is really a fantastic tool.
> > alas, I cannot use my ESP32-S2 as the platform is missing:(
> > I hope the implementation arrives soon ;)
> 
> I "solved" this problem by programming in Arduino for the boards that are not supported But pain is there, and instead of native ESPHome protocol I am using MQTT of course At least I can learn more about C and MQTT while waiting for ESPHome supporting new boards on arm64 ;-)

You can also use ESPHOME installed on a normal x64 linux as a workaround.

**Comment by homonto** [comment]
> You can also use ESPHOME installed on a normal x64 linux as a workaround.

I thought about it but I have at home only arm based computers: RPi and Macbooks
So not even VM will help

**Comment by rbray89** [comment]
Same. Been running on a docker instance on an x86 -64 machine to get around this. With RPI4 being aarch64, you'd think this would be more urgentl.

**Comment by Jason2866** [comment]
Toolchains are directly provided (since a while) from espressif to the Platformio Registry. The feature request is better placed in the corresponding espressif github.
https://registry.platformio.org/tools/espressif/toolchain-xtensa-esp32/compatibility
https://github.com/espressif/crosstool-NG.git

**Comment by homonto** [comment]
> Toolchains are directly provided (since a while) from espressif to the Platformio Registry. The feature request is better placed in the corresponding espressif github.
> https://registry.platformio.org/tools/espressif/toolchain-xtensa-esp32/compatibility
> https://github.com/espressif/crosstool-NG.git

you are saying: there is not toolchain for arm64? so how is this working on Arduino?

**Comment by Jason2866** [comment]
I just say if you want to use a toolchain from espressif for platformio you have to ask espressif. Not more not less. Idk if there is a toolchain or not.

**Comment by maedball** [comment]
https://github.com/espressif/crosstool-NG/issues/21

Opened ;-)

**Comment by quentinmit** [comment]
As a workaround you can invoke platformio or esphome with `linux32`, which will cause platformio to think it's running in 32-bit mode:

```
# If not already enabled on your system
sudo dpkg --add-architecture armel
sudo apt update
sudo apt install libc6:armel libstdc++6:armel
# Now build hello world
mkdir /tmp/test
cd /tmp/test
pio project init --board esp32-c3-devkitm-1
linux32 pio run
```

**Comment by Jason2866** [comment]
I have builded a aarch64 toolchain set for Linux aarch64 useable with Platformio
https://github.com/Jason2866/crosstool-NG/releases

**Comment by platima** [comment]
> I have builded a aarch64 toolchain set for Linux aarch64 useable with Platformio https://github.com/Jason2866/crosstool-NG/releases

You are an amazing person <3

**Comment by GNAKACH** [comment]
Brilliant! A million thanks!

**Comment by platima** [comment]
Note had to set 
```
  platformio_options:
    board_build.flash_mode: dio
```

**Comment by Jason2866** [comment]
@platima Option is in no way related to the toolchain. "Just" the board

**Comment by platima** [comment]
> @platima Option is in no way related to the toolchain. "Just" the board

Oh, would not all esp32s2's be flashed that same way then? I thought it was a chip specification, not board!

Cheers

**Comment by Jason2866** [comment]
No, since it is a option for the flash chip and how it is connected to the esp32s2
There are different options for the flash chip and how it is connected.

**Comment by platima** [comment]
Good to know! Thanks mate

**Comment by Jason2866** [comment]
Supported. Use latest Platform-espressif32 release.

**Comment by platima** [comment]
Sweet! Thansk
